### PR TITLE
Wip/superm1/device id fixes

### PR DESCRIPTION
**Pull request by superm1**

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -957,17 +957,36 @@ fu_util_get_history (FuUtilPrivate *priv, gchar **values, GError **error)
...
 
+static FwupdDevice*
+fu_util_get_device_or_prompt (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	FwupdDevice *dev = NULL;
+
+	/* get device to use */
+	if (g_strv_length (values) >= 1) {
+		if (g_strv_length (values) > 1)
+			for (guint i=1; i < g_strv_length (values); i++)
```
Comment (hughsie) [inline src/fu-util.c]: spaces around the = pls

File: src/fu-util.c [hunk 2]
```diff
@@ -957,17 +957,36 @@ fu_util_get_history (FuUtilPrivate *priv, gchar **values, GError **error)
...
+				g_debug ("Ignoring extra input %s", values[i]);
+		dev = fwupd_client_get_device_by_id (priv->client, values[0],
+						     NULL, error);
+		if (dev != NULL)
+			return dev;
+		g_print ("%s\n", (*error)->message);
```
Comment (hughsie) [inline src/fu-util.c]: Kinda nasty. A local `GError *error_local = NULL` would be a better way to do this imho.

File: src/fu-util.c [hunk 3]
```diff
@@ -1250,19 +1269,17 @@ static gboolean
 fu_util_get_results (FuUtilPrivate *priv, gchar **values, GError **error)
 {
 	g_autofree gchar *tmp = NULL;
-	g_autoptr(FwupdDevice) dev = NULL;
+	g_autoptr(FwupdDevice) indev = NULL;
```
Comment (hughsie) [inline src/fu-util.c]: what does `indev` mean?

Comment (superm1) [reply src/fu-util.c]: I just didn't want to reuse the g_autoptr twice. Will it handle freeing in between uses? If so I'll just keep it as one pointer.

File: src/fu-util.c (hunk 2 above)
Comment (superm1) [reply src/fu-util.c]: Ok. Will adjust to a local error.

File: src/fu-util.c (hunk 3 above)
Comment (hughsie) [reply src/fu-util.c]: Nahh, the g_autoptr is fine, I just didn't like the indev name.

**Comment by superm1** [comment]
Other thing I noticed was ctrl-c doesn't work at the prompt either. It seemed like you set up the signal handler for sigint right though so I wasn't too sure why that was busted.

File: src/fu-util.c [hunk 4]
```diff
@@ -957,17 +957,36 @@ fu_util_get_history (FuUtilPrivate *priv, gchar **values, GError **error)
...
 
+static FwupdDevice*
+fu_util_get_device_or_prompt (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	FwupdDevice *dev = NULL;
+
+	/* get device to use */
+	if (g_strv_length (values) >= 1) {
+		g_autoptr(GError) error_local = NULL;
+		if (g_strv_length (values) > 1)
```
Comment (hughsie) [inline src/fu-util.c]: I think you need `{``s around  this if - it's multiline and so a bit confusing.

**Comment by hughsie** [comment]
One tiny nit, and then LGTM. Thanks.

File: src/fu-util.c (hunk 4 above)
Comment (superm1) [reply src/fu-util.c]: I'll double check, but even if it works style wise you'd rather they be there?

Comment (hughsie) [reply src/fu-util.c]: it just looked odd to me to see a "single line" if with a for loop nested below.

**Pull request merged**